src/core/qgsvectorfilewriter.cpp

```diff
@@ -2744,6 +2744,34 @@ QMap<QString, QString> QgsVectorFileWriter::ogrDriverList()
   return resultMap;
 }
 
+QString QgsVectorFileWriter::driverForExtension( const QString &extension )
+{
+  QString ext = extension.trimmed();
+  if ( ext.isEmpty() )
+    return QString();
+  QgsApplication::registerOgrDrivers();
+  int const drvCount = OGRGetDriverCount();
+
+  for ( int i = 0; i < drvCount; ++i )
+  {
+    OGRSFDriverH drv = OGRGetDriver( i );
+    if ( drv )
+    {
+      QString drvName = OGR_Dr_GetName( drv );
+      if ( OGR_Dr_TestCapability( drv, "CreateDataSource" ) != 0 )
```

Comment:
Why restricting to drivers with creation capabilities ? The doc of the function doesn't suggest this restriction.

(side note: with GDAL 2.0, by using the GDAL API rather than the "old" OGR one, you can query GDALGetMetadata(hDriver, GDAL_DMD_EXTENSIONS), but there might not be a full identity between what QGIS has registered for extensions in its driverMetadata() and the one of the GDAL/OGR drivers themselves )